packages/liveblocks-react/src/comments/CommentsRoom.ts

```diff
@@ -93,166 +118,262 @@ export type RoomThreads<TThreadMetadata extends BaseMetadata> =
+  public request: RequestInfo<Data> | undefined; // Stores the currently active revalidation request
+  public mutation: MutationInfo | undefined; // Stores the start and end time of the currently active mutation
 
-  function startMutation() {
-    pollingHub.threads.stop();
-    numberOfMutations++;
+  constructor() {
+    super();
   }
 
-  const pollingHub = {
-    // TODO: If there's an error, it will currently infinitely retry at the current polling rate → add retry logic
-    threads: makePoller(revalidateThreads),
-  };
-  let unsubscribeRealtimeEvents: (() => void) | undefined;
-  let unsubscribeRealtimeConnection: (() => void) | undefined;
-  let realtimeClientConnected = false;
-
-  function getPollingInterval() {
-    return realtimeClientConnected
-      ? POLLING_INTERVAL_REALTIME
-      : POLLING_INTERVAL;
+  /* -------------------------------------------------------------------------------------------------
+   * Cache Getter/Setter
+   * -----------------------------------------------------------------------------------------------*/
+  get cache(): State<Data> | undefined {
+    return this._cache;
   }
 
-  function ensureThreadsAreLoadedForMutations() {
-    const state = store.get();
-    if (state.isLoading || state.error) {
-      throw new Error(
-        "Cannot update threads or comments before they are loaded"
-      );
-    }
-    return state.threads;
+  set cache(value: State<Data> | undefined) {
+    this._cache = value;
+    const event = new CustomEvent("cacheupdate");
+    this.dispatchEvent(event);
   }
 
-  async function revalidateThreads() {
-    pollingHub.threads.pause();
+  /* -------------------------------------------------------------------------------------------------
+   * Event Listener
+   * -----------------------------------------------------------------------------------------------*/
+  addEventListener(
+    key: "cacheupdate",
+    callback: EventListenerOrEventListenerObject | null,
+    options?: boolean | AddEventListenerOptions | undefined
+  ): void;
+
+  addEventListener(
+    key: string,
+    callback: EventListenerOrEventListenerObject | null,
+    options?: boolean | AddEventListenerOptions | undefined
+  ): void {
+    super.addEventListener(key, callback, options);
+  }
 
-    if (numberOfMutations === 0) {
-      if (fetchThreadsPromise === null) {
-        fetchThreadsPromise = room.getThreads();
-      }
-      setThreads(await fetchThreadsPromise);
-      fetchThreadsPromise = null;
-    }
+  removeEventListener(
+    type: "cacheupdate",
+    callback: EventListenerOrEventListenerObject | null,
+    options?: boolean | EventListenerOptions | undefined
+  ): void;
 
-    pollingHub.threads.resume();
+  removeEventListener(
+    type: string,
+    callback: EventListenerOrEventListenerObject | null,
+    options?: boolean | EventListenerOptions | undefined
+  ): void {
+    super.removeEventListener(type, callback, options);
   }
+}
```

Comment:
These are likely needed to enforce type-safety of the params here since they're just calling `super`'s method, right? None of that code is needed if you use the event source API instead, and it's type-safe automatically.